appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChartData.java

```diff
@@ -0,0 +1,137 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.editor.simple.SimpleEditor;
+import com.google.gwt.resources.client.ImageResource;
+import com.google.gwt.user.client.ui.Image;
+import com.google.gwt.user.client.ui.InlineHTML;
+import com.google.gwt.user.client.ui.SimplePanel;
+
+public abstract class MockChartData extends MockVisibleComponent {
+    private static final String PROPERTY_COLOR = "Color";
+    private static final String PROPERTY_LABEL = "Label";
+    private static final String PROPERTY_PAIRS = "ElementsFromPairs";
+
+    // Represents the Chart data icon
+    private Image iconWidget;
+
+    protected MockChart chart;
+    protected MockChartDataModel chartDataModel;
+
+    private String currentElements = "";
+
+    /**
+     * Creates a new instance of a Mock Chart Data component.
+     *
+     * @param editor editor of source file the component belongs to
+     * @param type  type string of the component
+     * @param icon  icon of the component
+     */
+    MockChartData(SimpleEditor editor, String type, ImageResource icon) {
+        super(editor, type, icon);
+
+        iconWidget = new Image(icon);
```

Comment:
Should an Image be put into a Panel, or is it fine to initialize the Image component itself?